torch/csrc/jit/codegen/cuda/runtime/memory.cu

```diff
@@ -0,0 +1,107 @@
+// Utility macro for this file
+#define DEVICE_INLINE __device__ inline
+
+#if (defined(__CUDA_ARCH__) && (__CUDA_ARCH__ >= 750))
+
+namespace Turing {
+
+namespace util {
+
+// Special utility for ldmatrix and cp_async
+DEVICE_INLINE unsigned toSmem(const void* ptr) {
```

Comment:
Utility for converting generic pointer to SMEM pointer in PTX. We should review vectorized load/stores with shared memory. SMEM memory movement PTX is only Global -> SMEM, SMEM -> Local, Local -> SMEM, and this is needed for these PTX instructions to provide the SMEM pointer.

Reply:
Added to comment. Thanks.